Plugins/Triage/Retrieve-Items.ps1

```diff
@@ -179,42 +227,134 @@ process{
+                        $FileStream.Close()
+                        $FileStream.Dispose()
+                        $IsLocked = $false
+
+                    } catch {
+
+                        $IsLocked = $true
+                        Break
+                    }
+                }
+
             } -ArgumentList $Item
         } 
 
-        Invoke-Command -Session $Session -ScriptBlock {
+        $Locked = Invoke-Command -Session $Session -ScriptBlock {$IsLocked}
+        #Process the $item based on IsLocked value
+
+        if (!$Locked){
+
+            Invoke-Command -Session $Session -ScriptBlock {
 
             #Create archive of Item and MetaData (separately)
             $ArchivePath = "C:\ProgramData\{0}.zip" -f (Split-Path $($args[1]) -Leaf)
             $Command_Compress = "C:\ProgramData\{0} a -pinfected -tzip {1} {2} {3}" -f ($($args[0]), $ArchivePath, $ExportPath, ($($args[1])))
 
             Invoke-Expression -Command $Command_Compress | Out-Null
 
-        } -ArgumentList (Split-Path $Installexe -Leaf), $Item
+            } -ArgumentList (Split-Path $Installexe -Leaf), $Item
+
+            #Copy specified archive to $Output
+            $ItemPath = "C:\ProgramData\{0}.zip" -f (Split-Path $Item -Leaf)
+
+            Copy-Item -Path $ItemPath -Destination "$Output\" -FromSession $Session -Force -ErrorAction SilentlyContinue
+
+            #Remove created files on remote machine as cleanup
+
+            Invoke-Command -Session $Session -ScriptBlock {
+
+                #Remove the archive
+
+                Remove-Item -Path $ArchivePath -Force 
+                    
+                #Remove the Metadata file
+
+                Remove-Item -Path $ExportPath -Force  
+
+            }
+        }
+
+        if ($Locked){
+
+            #Deploy Velociraptor
+
+            $Test_Velo = Invoke-Command -Session $Session -ScriptBlock {Get-Item ("C:\ProgramData\{0}" -f $($args[0])) -ErrorAction SilentlyContinue} -ArgumentList (Split-Path $Velo_exe -Leaf)
 
-        #Copy specified archive to $Output
-        $ItemPath = "C:\ProgramData\{0}.zip" -f (Split-Path $Item -Leaf)
+            if (!$Test_Velo){
 
-        Copy-Item -Path $ItemPath -Destination "$Output\" -FromSession $Session -Force -ErrorAction SilentlyContinue
+                try{
 
-        #Remove created files on remote machine as cleanup
+                    Copy-Item -Path $Velo_exe -Destination "C:\ProgramData" -ToSession $Session -ErrorAction Stop 
 
-        Invoke-Command -Session $Session -ScriptBlock {
+                }catch {
 
-            #Remove the archive
+                    Throw ("Locked file detected at {0}, but could not deploy Velociraptor for retrieval. Quitting..." -f $Item)
+                }
+
+            }
+
+            #Copy $Item with Velociraptor, compress, and copy back to origin machine
+
+            #Collect $Item
+            $FinalPath = "C:\ProgramData\{1}_{0}" -f $Session.ComputerName, (Split-Path $Item -Leaf)
```

Comment:
Did you intend for this to be `C:\ProgramData\File.Ext_ComputerName`?

Reply:
I did. Not sure it is the best thing to do, but i wanted to be able to see the exact item i was getting in my naming convention. I may change this if it causes issues in the future. Open to other thoughts if it would be a bad idea to do.